pkg/rc/replication_controller.go

```diff
@@ -316,6 +308,72 @@ func (rc *replicationController) addPods(current types.PodLocations, eligible []
 	return nil
 }
 
+func (rc *replicationController) doNotEnoughNodes(currentNodes []types.NodeName, eligibleNodes []types.NodeName, numScheduledNodes int, txn *auditingTransaction) error {
+	var err error = nil
+	if rc.AllocationStrategy == fields.CattleStrategy {
+		err = rc.doNotEnoughCattleNodes(currentNodes, eligibleNodes, numScheduledNodes)
+	} else {
+		err = rc.doNotEnoughPetNodes(currentNodes, eligibleNodes, numScheduledNodes)
+	}
+
+	// commit any queued operations
+	ok, resp, txnErr := txn.Commit(rc.txner)
+	switch {
+	case txnErr != nil:
+		return txnErr
+	case !ok:
+		return util.Errorf("could not schedule pods due to transaction violation: %s", transaction.TxnErrorsToString(resp.Errors))
+	}
+
+	return err
+}
+
+func (rc *replicationController) doNotEnoughCattleNodes(currentNodes []types.NodeName, eligibleNodes []types.NodeName, numScheduledNodes int) error {
+	rcStatus, _, err := rc.rcStatusStore.Get(rc.ID())
+	isNoNodeTransferInProgress := (err != nil && statusstore.IsNoStatus(err)) ||
+		(err == nil && rcStatus.NodeTransfer == nil)
+	if isNoNodeTransferInProgress {
+		newNodes, err := rc.scheduler.AllocateNodes(rc.Manifest, rc.NodeSelector, len(currentNodes)+1)
+		if err != nil || len(newNodes) < 1 {
```

Comment:
We need to capture this error message if it's not nil.

Reply:
Will do